### ISSUE TITLE
[Bug]: Both Local and Individuous API don't work. Pulling Null of Video

### ISSUE THREAD

**Issue by kasselvania** [FreeTubeApp/FreeTube#5534]
### Guidelines

- [X] I have encountered this bug in the [latest release of FreeTube](https://github.com/FreeTubeApp/FreeTube/releases).
- [X] I have encountered this bug in the [official downloads of FreeTube](https://github.com/FreeTubeApp/FreeTube#official-downloads).
- [X] I have searched the issue tracker for [open](https://github.com/FreeTubeApp/FreeTube/issues?q=is%3Aopen+is%3Aissue) and [closed](https://github.com/FreeTubeApp/FreeTube/issues?q=is%3Aissue+is%3Aclosed) issues that are similar to the bug report I want to file, without success.
- [X] I have searched the [documentation](https://docs.freetubeapp.io/) for information that matches the description of the bug I want to file, without success.
- [X] This issue contains only one bug.

### Describe the bug

Whenever attempting to open a video, I am greeted by messages that say:

"Local API Error: Type Error: Cannot read properties of null (reading 'reason'). Then, Invidious API error: Error: Missing harsh key "videoDetails"

I cannot get any videos to play
<img width="1000" alt="Screenshot 2024-08-06 at 1 42 26 PM" src="https://github.com/user-attachments/assets/057c68b1-a027-42f4-9736-daeadd11ea01">

### Expected Behavior

I would expect to click the video link and have the video play.

### Issue Labels

API issue

### FreeTube Version

v0.21.3 Beta

### Operating System Version

macOS Sonoma 14.6

### Installation Method

.dmg

### Primary API used

Local API

### Last Known Working FreeTube Version (If Any)

v0.21.1 Beta

### Nightly Build

- [ ] I have encountered this bug in the latest [nightly build](https://docs.freetubeapp.io/development/nightly-builds).

**Comment by Gorrrg** [comment]
I've found your bugfix:
> **Last Known Working FreeTube Version (If Any)**
> v0.21.1 Beta
Change the last line to v0.21.3 beta. 😅

**Comment by kasselvania** [comment]
I have actually figured out that my VPN, Private Internet Access, doesn't play well with FreeTube. Can't figure out why.

**Comment by qwtoe** [comment]
I met the same error

**Comment by qwtoe** [comment]
> I met the same error

i found out a solution, just simply adjust external player in Settings:

![image](https://github.com/user-attachments/assets/9a4f5e7b-bed2-464d-ac48-bbcc688ef842)

**Comment by absidue** [comment]
@kasselvania Can you try opening the devtools with <kbd>Cmd</kbd>+<kbd>Shift</kbd>+<kbd>I</kbd> and look at the console tab? Maybe that has more about details about the error messages. Additionally does this happen with every video or only specific ones?

**Comment by kasselvania** [comment]
> @kasselvania Can you try opening the devtools with Cmd+Shift+I and look at the console tab? Maybe that has more about details about the error messages. Additionally does this happen with every video or only specific ones?

Honestly? There is no way to tell what kind of video will work, as apposed to others. If I open the developer tools, I see stuff like this screenshot.

<img width="553" alt="Screenshot 2024-08-15 at 7 40 30 PM" src="https://github.com/user-attachments/assets/44928365-fe0b-4ad0-af50-79a2420c2c8e">


@Ammako I will try logging in and see if that corrects my issue! Thanks for the suggestion!

**Comment by absidue** [comment]
@Ammako Even if that is the case, I think we can all agree that it would be better to figure out what YouTube returns in that case so that FreeTube can show a proper error message instead of breaking like it does now.

**Comment by absidue** [comment]
@kasselvania Interesting, I'm guessing you got a different error messages in the toasts/popups this time (something a long the line of failed to fetch)? So this time it seems like you got a different problem, the last red message says "ERR_NAME_NOT_RESOLVED`, which means it failed to resolve the DNS name to an IP address or in less technical terms it failed to figure out what IP address (combination of numbers) is associated to `www.youtube.com`.

Try connecting and disconnecting from your VPN, to solve the problem you got this time. As for the original problem, it seems like that is still somewhat of a mystery.

**Comment by kasselvania** [comment]
@absidue i still get the error in the main window like above not pulling Null from the video, and the video stating it’s from Dec 31, 1960, with no data presented. I usually need to quit Freetube, turn off my VPN, and then reopen freetube WITHOUT my VPN for a few videos for it to work again. Then, once it’s running, for a video or two, I can add the VPN back into the mix again. This problem is pretty frequent, though.

**Comment by Victor239** [comment]
Yeah I get this problem consistently when on my VPN also, it falls back to Invidious.

**Comment by absidue** [comment]
If someone could capture what the response from YouTube looks like when that error occurs, that would help a lot in trying to figure out what is going on, as I've not been able to reproduce this on various VPN IP addresses.

Here is how you could do that:

1. Open the developer tools by pressing <kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>I</kbd> or <kbd>Command</kbd>+<kbd>Shift</kbd>+<kbd>I</kbd> on macOS
2. Switch to the `Network` tab
3. In the text box that says `Filter` type `/player` (you won't see any results yet, that will come later)
4. Open a video (leave the devtools open otherwise they won't capture any data)
5. After the error occurs take a look at the list in the `Network` tab, you should see two entries whose names start with `player?`
6. Repeat these steps for each of the requests (there should be two)
    * Right click on the request
    * Go to `Copy `
    * The click on `Copy Response`
    * Paste the response into a text file.
7. Upload the files here or if you don't feel comfortable doing that you can also send them to my email address `git (at) absidue.me`.

**Comment by EugeneDubuk** [comment]
@absidue 
[#5534.zip](https://github.com/user-attachments/files/16830233/5534.zip)

**Comment by absidue** [comment]
Okay so as expected the cause of the error is that you are using a VPN IP address that is blocked by YouTube. FreeTube needs to display a better error message when that happens but the only way to prevent it happening is to use a different VPN IP address that isn't blocked.

**Comment by EugeneDubuk** [comment]
How about adding optional login box in Freetube for such cases?

**Comment by absidue** [comment]
Firstly that would go against FreeTube's primary goal of being a privacy friendly app and secondly despite the error message suggesting you login, I would really not recommend doing that on a blocked IP even in your web browser unless you want your Google account blocked (people that have tried to use logging in to bypass IP blocks have had their accounts blocked ).

**Comment by absidue** [comment]
> source on people getting their google account blocked for logging in on web/official youtube app from these kind of vpn ips?

I think you may have misinterpreted the message or I wasn't clear enough in my writing.

**Comment by absidue** [comment]
The resolution to this issue will be for FreeTube to show a better error message that explains that you need to change to a new IP address, something along the lines of: "YouTube has blocked the IP address that you are using, if you are using a VPN try changing to a different server in your VPN settings, if you are not using a VPN or a proxy, you will have to start using one"

**Comment by Victor239** [comment]
There's currently a setting to Proxy Videos Through Invidious, which overrides API preference. It would be helpful if this could be used as a fallback in case Local API is blocked due to VPN use.

**Comment by absidue** [comment]
@Victor239 FreeTube already falls back to the Invidious API if the local API fails and vice versa, as long as you haven't turned off the `Fallback to Non-Preferred Backend on Failure` setting in the general settings (it's on by default), the Proxy Videos Through Invidious setting is unrelated.

You can see the fallback in action in the screenshot in the description of the issue, they were just unlucky enough to also be using an Invidious instance that was broken which is why it still didn't work even after falling back to the Invidious API.

**Comment by EugeneDubuk** [comment]
One more idea. 
There is an option in `...` > `Open YouTube Embedded Player`.
It opens video in my browser and it works perfectly.
But what would be much convenient, if I could set a `default action` setting to open all videos in browser by default, without clicking on `...`. 
And the second, video have to be marked as `WATCHED` after opening it in browser.

**Comment by Mind777** [comment]
I am frequently getting this error also.

**Comment by Victor239** [comment]
> One more idea. 
> There is an option in `...` > `Open YouTube Embedded Player`.
> It opens video in my browser and it works perfectly.
> But what would be much convenient, if I could set a `default action` setting to open all videos in browser by default, without clicking on `...`. 
> And the second, video have to be marked as `WATCHED` after opening it in browser.

I like this idea as well for times when FreeTube suddenly stops playing videos due to YouTube changes. Another useful option would be to always open external player as a default action.

**Comment by andriuhatm** [comment]
Hmm was getting this error with Surfshark VPN enabled as well. Seems to work fine with it off.

**Comment by Jiejue233** [comment]
Local API has developed the same failure on my end, and according to my own investigation, this `Local API Error - TypeError: Cannot read properties of null (reading 'reason')` failure typically happens when the HTTP request is returning error 429 (too many requests), namely, the IP block of VPN. 

SO:
- For the app side, there is really nothing to handle, however, making the error message more specific helps
- For the users, you will really need to change the proxy/region that you are using, or change the provider can also solve this problem

**Comment by Yolakalemowa** [comment]
> Local API has developed the same failure on my end, and according to my own investigation, this `Local API Error - TypeError: Cannot read properties of null (reading 'reason')` failure typically happens when the HTTP request is returning error 429 (too many requests), namely, the IP block of VPN.
> 
> SO:
> 
>     * For the app side, there is really nothing to handle, however, making the error message more specific helps
> 
>     * For the users, you will really need to change the proxy/region that you are using, or change the provider can also solve this problem

Hello, I'm not even on VPN but still local api not working.

"Error: This helps protect our community. Learn more"

**Comment by absidue** [comment]
@Yolakalemowa That error is coming from Invidious, as is explained in the "maintainer edit" section of the top post, you need to stop using Invidious for the moment, until they have found a solution on their end.

**Comment by Yolakalemowa** [comment]
> @Yolakalemowa That error is coming from Invidious, as is explained in the "maintainer edit" section of the top post, you need to stop using Invidious for the moment, until they have found a solution on their end.

sorry bit of a noob here. But yes I thought I wasn't using Invid. I chose preferred api backend as local API not Invid.

Or am I miunderstanding?

**Comment by Yolakalemowa** [comment]
> @Yolakalemowa That error is coming from Invidious, as is explained in the "maintainer edit" section of the top post, you need to stop using Invidious for the moment, until they have found a solution on their end.

Hi... what does this window mean? 

![image](https://github.com/user-attachments/assets/55da0f6d-a7f5-4a86-9ab3-7d4af3fd189a)

**Comment by CaptainMorgan12** [comment]
Environment: freetube v0.21.3 Beta .deb amd64 or flatpak install, Ubuntu 24.04.1 LTS, custom t2 kernel Linux 6.11.0-1-t2-noble, MacPro 7,1. Internet works, Firefox works, 

Getting the same issues recently.

![Screenshot from 2024-09-19 10-58-54](https://github.com/user-attachments/assets/b76c9ed9-d3b0-4976-a9c3-de46bc528671)

The way it shows up is that videos cannot be played, while subscription image previews show fine:

![image](https://github.com/user-attachments/assets/dc9bc150-78f4-4105-b677-30d1babbd70a)

**Comment by Jiejue233** [comment]
A note for everyone commented below: **The "this helps protect our community" or "this content isn't available" is *not* a freetube error**
This is because the **Invidious API** (an open-sourced API used for anonymous access to youtube) is not working due to the recent update by youtube, this update is blocking every detected "bot connection"
Hence, everything related to Invidious should be reported to the [invidious API github](https://github.com/iv-org/invidious/issues) though you don't have to, since [this](https://github.com/iv-org/invidious/issues/4734) issue is already done so.
Any further report like "this is also happened to me" is unwelcome. If you are reporting anything, see through the report and if none of the reports covers your case (in general sense), then your report is meaningful
And finally @Yolakalemowa in your case, you can check if 
![image](https://github.com/user-attachments/assets/550c16ab-e8f9-4da2-82d3-23f071a12767)
these two options are like this, this helps prevent auto fallback to Invidious API if local API fails

**Comment by absidue** [comment]
@CaptainMorgan12 As mentioned earlier in the thread "Cannot read properties of null" is the error that happens with the local API when YouTube has blocked your IP address (FreeTube is failing to read the error message from YouTube's response but it is getting the same error message from YouTube that blocked Invidious instances get).

**Comment by Yolakalemowa** [comment]
> A note for everyone commenting below: **The "this helps protect our community" or "this content isn't available" is _not_ a freetube error** This is because the **Invidious API** (an open-sourced API used for anonymous access to youtube) is not working due to the recent update by youtube, this update is blocking every detected "bot connection" Hence, everything related to Invidious should be reported to the [invidious API github](https://github.com/iv-org/invidious/issues) though you don't have to, since [this](https://github.com/iv-org/invidious/issues/4734) issue is already done so. Any further report like "this is also happened to me" is unwelcome. If you are reporting anything, see through the report and if none of the reports covers your case (in general sense), then your report is meaningful And finally @Yolakalemowa in your case, you can check if ![image](https://private-user-images.githubusercontent.com/78036442/369123945-550c16ab-e8f9-4da2-82d3-23f071a12767.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY5MDc2NzEsIm5iZiI6MTcyNjkwNzM3MSwicGF0aCI6Ii83ODAzNjQ0Mi8zNjkxMjM5NDUtNTUwYzE2YWItZThmOS00ZGEyLTgyZDMtMjNmMDcxYTEyNzY3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTIxVDA4MjkzMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZmNjkzMGY1ZjNjNzcyMmQwMWE0YTM2NGE2ZDkyZWI4YzJlZWM0YWRkNzc4ZmNhMDAxOTBlMDVmZGI5NjgzZDgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.mMGNgYWPiLkzuxjv7ZqTC06Caf0LavdYL9yaQ68hIpg) these two options are like this, this helps prevent auto fallback to Invidious API if local API fails

Hi,

Sorry to bother you again but I tried the above recommendations and still not working:

![image](https://github.com/user-attachments/assets/4bdd13ca-ed3e-45b5-8396-7cc2aa7b7daf)

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Is anyone here using a VPN service that is free to use and getting IP blocked? If so please provide the following details 

- Name of VPN service
- Server you are getting blocked on

**Comment by VideoGames1000** [comment]
I've been getting this error without even having a VPN installed at seemingly-random, which is lovely.
Should I be concerned about this?

**Comment by absidue** [comment]
@VideoGames1000 Are you just using FreeTube or also doing things like downloading videos with yt-dlp? According to this comment here, using yt-dlp a lot can also cause the errors: https://github.com/FreeTubeApp/FreeTube/issues/5750#issuecomment-2378313756

**Comment by VideoGames1000** [comment]
Casual (if fairly frequent) Freetube use, I've only downloaded one or two videos total since this started happening.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
> Is anyone here using a VPN service that is free to use and getting IP blocked? If so please provide the following details
> 
>     * Name of VPN service
> 
>     * Server you are getting blocked on

Based on getting 0 responses on this request I assume that nobody that is using a free VPN service is getting hit by this

Requesting the following info, please provide the following info if you are using a **paid** VPN service and getting IP blocked:
- Name of the VPN service
- Serve you are getting blocked on

**Comment by EugeneDubuk** [comment]
@efb4f5ff-1298-471a-8973-3d47447115dc 
I use my private EC in AWS. Can provide you with an account if you wish.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
@EugeneDubuk Sure, how do you wish to send over the details?

**Comment by EugeneDubuk** [comment]
@efb4f5ff-1298-471a-8973-3d47447115dc
Give me an email where I can send it.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
> @efb4f5ff-1298-471a-8973-3d47447115dc
> Give me an email where I can send it.

@EugeneDubuk 

- abiding_supply793@simplelogin.com

Email alias will be disabled after receiving the details.

**Comment by EugeneDubuk** [comment]
@efb4f5ff-1298-471a-8973-3d47447115dc
Sent an email.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Will respond when im behind my machine can take a few days though

**Comment by Yolakalemowa** [comment]
> > A note for everyone commenting below: **The "this helps protect our community" or "this content isn't available" is _not_ a freetube error** This is because the **Invidious API** (an open-sourced API used for anonymous access to youtube) is not working due to the recent update by youtube, this update is blocking every detected "bot connection" Hence, everything related to Invidious should be reported to the [invidious API github](https://github.com/iv-org/invidious/issues) though you don't have to, since [this](https://github.com/iv-org/invidious/issues/4734) issue is already done so. Any further report like "this is also happened to me" is unwelcome. If you are reporting anything, see through the report and if none of the reports covers your case (in general sense), then your report is meaningful And finally @Yolakalemowa in your case, you can check if ![image](https://private-user-images.githubusercontent.com/78036442/369123945-550c16ab-e8f9-4da2-82d3-23f071a12767.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY5MDc2NzEsIm5iZiI6MTcyNjkwNzM3MSwicGF0aCI6Ii83ODAzNjQ0Mi8zNjkxMjM5NDUtNTUwYzE2YWItZThmOS00ZGEyLTgyZDMtMjNmMDcxYTEyNzY3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTIxVDA4MjkzMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZmNjkzMGY1ZjNjNzcyMmQwMWE0YTM2NGE2ZDkyZWI4YzJlZWM0YWRkNzc4ZmNhMDAxOTBlMDVmZGI5NjgzZDgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.mMGNgYWPiLkzuxjv7ZqTC06Caf0LavdYL9yaQ68hIpg) these two options are like this, this helps prevent auto fallback to Invidious API if local API fails
> 
> Hi,
> 
> Sorry to bother you again but I tried the above recommendations and still not working:
> 
> ![image](https://private-user-images.githubusercontent.com/41283612/369606083-4bdd13ca-ed3e-45b5-8396-7cc2aa7b7daf.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjc5NDMwMjYsIm5iZiI6MTcyNzk0MjcyNiwicGF0aCI6Ii80MTI4MzYxMi8zNjk2MDYwODMtNGJkZDEzY2EtZWQzZS00NWI1LTgzOTYtN2NjMmFhN2I3ZGFmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDAzVDA4MDUyNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJlNmFmZjc2N2JmNTBiN2MyNWFjMmZjNzFhNDU1NTA4NzIyOGE2OTI2NDlmOWE1YWY1MzJjMTRkNGMxYzNiZTImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.WPM8WMhspZ9GaLY90nUkmcFD_IxIPizIktsJKlijirs)

Hi I'm still getting an error on local api ... local api error: error: please sign in.

Is there any suggestion to fix it? Much appreciated

**Comment by absidue** [comment]
@Yolakalemowa Make sure you are using the local API and have Player Settings -> Proxy Videos through Invidious disabled. If both of those things are already the case, then you need to change to a different IP address (e.g. different VPN server or proxy server), the next release will show a better error message but as the block is against the IP address and not FreeTube there isn't much we can do about it.

**Comment by Yolakalemowa** [comment]
> @Yolakalemowa Make sure you are using the local API and have Player Settings -> Proxy Videos through Invidious disabled. If both of those things are already the case, then you need to change to a different IP address (e.g. different VPN server or proxy server), the next release will show a better error message but as the block is against the IP address and not FreeTube there isn't much we can do about it.

thanks but that's the weird thing. I have those settings as described, and I'm not even using a vpn :S

**Comment by absidue** [comment]
@Yolakalemowa As you seem to be saying that your home IP address is blocked, I would suggest using a VPN or a proxy for the moment.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
> @efb4f5ff-1298-471a-8973-3d47447115dc
> Sent an email.

@EugeneDubuk yesterday I found an server that was IP blocked. The account isnt needed anymore. Thankyou for providing it anyway!
